homeassistant/components/tado/device_tracker.py

```diff
@@ -49,27 +49,21 @@ def get_scanner(hass: HomeAssistant, config: ConfigType) -> TadoDeviceScanner |
 class TadoDeviceScanner(DeviceScanner):
     """Scanner for geofenced devices from Tado."""
 
-    def __init__(self, hass, config):
+    def __init__(self, hass: HomeAssistant, config) -> None:
         """Initialize the scanner."""
         self.hass = hass
-        self.last_results = []
+        self.last_results: list[str] = []
 
         self.username = config[CONF_USERNAME]
         self.password = config[CONF_PASSWORD]
+        self.fallback = CONST_OVERLAY_TADO_DEFAULT
 
         # The Tado device tracker can work with or without a home_id
         self.home_id = config[CONF_HOME_ID] if CONF_HOME_ID in config else None
 
-        # If there's a home_id, we need a different API URL
-        if self.home_id is None:
-            self.tadoapiurl = "https://my.tado.com/api/v2/me"
-        else:
-            self.tadoapiurl = "https://my.tado.com/api/v2/homes/{home_id}/mobileDevices"
-
-        # The API URL always needs a username and password
-        self.tadoapiurl += "?username={username}&password={password}"
-
-        self.websession = None
+        self.tadoconnector = TadoConnector(
```

Comment:
Why don't we set up a `ScannerEntity` from the config entry and use the existing connector already stored in hass.data?

https://github.com/home-assistant/core/blob/c9ba95ea1e3b871549161fa8ee730fb7af81130d/homeassistant/components/tado/__init__.py#L91-L92

Reply:
I was also discussing this with Erwin on Discord, but we first have to get rid of the YAML before we can move to async_setup_entry

Reply:
Just import it to an import step in the config flow. It's not that big of a task, I think.

Reply:
That's true, but I was in doubt if we should first do this PR and then do the import, or the other way around

Reply:
I think the current approach in the PR doesn't look good and will make it harder to coordinate the api calls in the integration. I suggest we go directly for the config entry approach.

Reply:
I'll work out the ScannerEntity.